reader.go

```diff
@@ -1514,15 +1525,8 @@ func (r *reader) read(ctx context.Context, offset int64, conn *Conn) (int64, err
 	var size int64
 	var bytes int64
 
-	const safetyTimeout = 10 * time.Second
-	deadline := time.Now().Add(safetyTimeout)
-	conn.SetReadDeadline(deadline)
-
 	for {
-		if now := time.Now(); deadline.Sub(now) < (safetyTimeout / 2) {
-			deadline = now.Add(safetyTimeout)
-			conn.SetReadDeadline(deadline)
-		}
+		conn.SetReadDeadline(time.Now().Add(r.maxSafetyTimeout))
```

Comment:
Maybe we could keep this simpler by moving this call before the loop? That way we only pay the cost of setting the deadline once per batch instead of once per message?

The code we had before was a bit too complex but attempted to reduce the cost of these `SetReadDeadline` calls.

Reply:
I agree, what I was thinking keeping it in every message, so that if it was small though, the batch still can continue. the problem I am afraid is that one don't know how big is the batch and it would be harder to give it the `SetReadDeadline ` a correct value? what do you think? correct me if i miss something